File: debug/src/main/java/com/sofar/debug/performance/monitor/MonitorCallback.java
package com.sofar.debug.performance.monitor;

public interface MonitorCallback {

  /**
   * @param frameRate 帧率
   */
  default void onFrameRate(int frameRate) {}

  /**
   * @param cpuRate cpu使用率
   */
  default void onCpuRate(float cpuRate) {}


  /**
   * @param count 线程数
   */
  default void onThreadCount(int count) {}

  /**
   * @param size 当前使用内存大小，单位MB
   */
  default void onMemory(float size) {}

  /**
   * @param count fd数量
   */
  default void onFDCount(int count) {}
}

File: tool/profiler/src/main/java/com/sofar/profiler/MonitorCallback.kt
package com.sofar.profiler

import androidx.annotation.MainThread
import com.sofar.profiler.activity.ActivityTimeInfo
import com.sofar.profiler.block.model.BlockInfo
import com.sofar.profiler.memory.model.MemoryInfo
import com.sofar.profiler.startup.model.StartupInfo

interface MonitorCallback {

  @MainThread
  fun onFrameRate(frameRate: Int) {
  }

  @MainThread
  fun onCpuRate(cpuRate: Float) {
  }

  @MainThread
  fun onThreadCount(count: Int) {
  }

  @MainThread
  fun onMemoryInfo(info: MemoryInfo) {
  }

  @MainThread
  fun onFDCount(count: Int) {
  }

  @MainThread
  fun onBatteryInfo(temperature: Float) {
  }

  @MainThread
  fun onTrafficInfo(info: String) {
  }

  @MainThread
  fun onAppStartInfo(info: StartupInfo) {
  }

  @MainThread
  fun onActivityStartInfo(info: ActivityTimeInfo) {
  }

  @MainThread
  fun onBlock(info: BlockInfo) {
  }
}
